src/components/SegmentedControl/Option/style.js

```diff
@@ -0,0 +1,95 @@
+import styled, { css } from 'styled-components';
+import { blue, blueLightest, boxShadow, gray, grayDark, grayLight } from '../../style/colors';
+import { borderRadius } from '../../style/borders';
+
+const getGridTemplateAreas = (iconPosition) => {
+  switch (iconPosition) {
+    case 'top':
+      return `
+        "icon"
+        "label"
+      `;
+
+    case 'bottom':
+      return `
+        "label"
+        "icon"
+      `;
+
+    case 'right':
+      return `"label icon"`;
+
+    case 'left':
+    default:
+      return `"icon label"`;
+  }
+};
+
+export const OptionStyled = styled.div`
+  display: flex;
+  align-items: center;
+  cursor: ${p => p.disabled ? 'not-allowed' : 'pointer'};
```

Comment:
I noticed you're using both `p` and `props`, not a major thing but I'd keep `props` for consistency